doc/en/components/grids/_shared/conditional-cell-styling.md

```diff
@@ -0,0 +1,684 @@
+    row.data.data['Title'].includes('President') ? '#adb5bd' : row.data.data['Title'].includes('Director') ?  '#ced4da' :
+    row.data.data['Title'].includes('Manager') ? '#dee2e6' :
+    row.data.data['Title'].includes('Lead') ? '#e9ecef' :
+    row.data.data['Title'].includes('Senior') ? '#f8f9fa' : null;
+
+public rowStyles = {
+    background: this.background,
+    'border-left': (row: RowType) => row.data.data['Title'] === 'CEO' || row.data.data['Title'].includes('President') ?
+        '2px solid' : null,
+    'border-color': (row: RowType) => row.data.data['Title'] === 'CEO' ? '#495057' : null,
+    color: (row: RowType) => row.data.data['Title'] === 'CEO' ? '#fff' : null
+};
+```
+
+```razor
+Add treegrid styles
+```
+
+<!-- Angular -->
+```html
+<!-- sample.component.html -->
+<igx-tree-grid #treeGrid [data]="data" [moving]="true" primaryKey="ID" foreignKey="ParentID"
+        width="100%" height="550px" [rowStyles]="rowStyles">
+</igx-tree-grid>
+```
+<!-- end: Angular -->
+
+```razor
+Add treegrid markup
+```
+
+<!-- ComponentEnd: TreeGrid -->
+
+<!-- ComponentStart: HierarchicalGrid -->
+
+```typescript
+// component.ts
+public rowStyles = {
+    background:(row: RowType) => row.data['HasGrammyAward'] ? '#eeddd3' : '#f0efeb',
+    'border-left': (row: RowType) => row.data['HasGrammyAward'] ? '2px solid #dda15e' : null
+};
+
+public childRowStyles = {
+    'border-left': (row: RowType) => row.data['BillboardReview'] > 70 ? '3.5px solid #dda15e' : null
+};
+```
+
+```razor
+Add hgrid styles
+```
+
+<!-- Angular -->
+```html
+<igx-hierarchical-grid  #hierarchicalGrid [data]="localdata" [autoGenerate]="false"
+        [height]="'580px'" [width]="'100%'" [rowStyles]="rowStyles">
+        <igx-row-island [key]="'Albums'" [autoGenerate]="false" [rowStyles]="childRowStyles">
+        </igx-row-island>  
+</igx-hierarchical-grid>
+```
+<!-- end: Angular -->
+
+```razor
+Add hgrid markup
+```
+<!-- ComponentEnd: HierarchicalGrid -->
+
+
+### Demo
+
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:620px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-rowStyles-sample"
+           github-src="{ComponentSample}/rowStyles-sample" >
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+
+
+<!-- ComponentStart: TreeGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-styles-sample"
+           github-src="{ComponentSample}/styles-sample" >
+</code-view>
+
+<!-- ComponentEnd: TreeGrid -->
+
+<!-- ComponentStart: HierarchicalGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-row-styles"
+           github-src="{ComponentSample}/row-styles" >
+           
+</code-view>
+
+<!-- ComponentEnd: HierarchicalGrid -->
```

Comment:
replace above 3 code-view blocks with just one code-view and remove component build flag because routing path to the sample will be the same for all grid samples in Sample Browser.